### ISSUE TITLE
HOw i can compile under Slackware(64) 14.2?

### ISSUE THREAD

**Issue by inukaze** [openmoh/openmohaa#25]
HI there i try to compile, but don't work, i do the follow :

```
cd /tmp
git clone https://github.com/openmoh/openmohaa
cd openmohaa
make

```

Output :
```
make[1]: se entra en el directorio '/tmp/openmohaa'

Building openmohaa in build/release-linux-x86_64:
  PLATFORM: linux
  ARCH: x86_64
  COMPILE_PLATFORM: linux
  COMPILE_ARCH: x86_64
  CC: cc

  CFLAGS:
    -MMD
    -Wall
    -fno-strict-aliasing
    -Wimplicit
    -Wstrict-prototypes
    -pipe
    -DUSE_ICON
    -I/usr/include/SDL
    -D_GNU_SOURCE=1
    -D_REENTRANT
    -DUSE_OPENAL
    -DUSE_CURL
    -DUSE_CURL_DLOPEN
    -DUSE_LOCAL_HEADERS
    -DNDEBUG
    -O3
    -fomit-frame-pointer
    -ffast-math
    -funroll-loops
    -falign-loops=2
    -falign-jumps=2
    -falign-functions=2
    -fstrength-reduce

  LDFLAGS:
    -ldl
    -lm

  Output:
    build/release-linux-x86_64/omohaaded.x86_64
    build/release-linux-x86_64/openmohaa.x86_64
    build/release-linux-x86_64/main/cgamex86_64opm.so
    build/release-linux-x86_64/main/gamex86_64opm.so
    build/release-linux-x86_64/main/uix86_64opm.so

make[2]: se entra en el directorio '/tmp/openmohaa'
DED_CC code/server/sv_bot.c
In file included from code/server/server.h:27:0,
                 from code/server/sv_bot.c:24:
code/server/../qcommon/q_shared.h:177:0: aviso: se redefinió "PAD"
 #define PAD(base, alignment) (((base)+(alignment)-1) & ~((alignment)-1))
 ^
code/server/../qcommon/q_shared.h:169:0: nota: esta es la ubicación de la definición previa
 #define PAD(x,y) (((x)+(y)-1) & ~((y)-1))
 ^
In file included from code/server/../qcommon/tiki.h:29:0,
                 from code/server/../qcommon/qcommon.h:27,
                 from code/server/server.h:28,
                 from code/server/sv_bot.c:24:
code/server/../qcommon/../tiki/tiki_shared.h:255:18: error: el campo ‘boneType’ tiene tipo de dato incompleto
  enum boneType_e boneType;
                  ^
In file included from code/server/../qcommon/qcommon.h:27:0,
                 from code/server/server.h:28,
                 from code/server/sv_bot.c:24:
code/server/../qcommon/tiki.h:117:25: error fatal: tiki_script.h: No existe el fichero o el directorio
compilación terminada.
Makefile:1683: fallo en las instrucciones para el objetivo 'build/release-linux-x86_64/ded/sv_bot.o'
make[2]: *** [build/release-linux-x86_64/ded/sv_bot.o] Error 1
make[2]: se sale del directorio '/tmp/openmohaa'
Makefile:847: fallo en las instrucciones para el objetivo 'targets'
make[1]: *** [targets] Error 2
make[1]: se sale del directorio '/tmp/openmohaa'
Makefile:841: fallo en las instrucciones para el objetivo 'release'
make: *** [release] Error 2
```

**Comment by mohabhassan** [comment]
Hi,
I didn't test building on Linux, but according to ley0k Linux builds work under eclipse. 
Not sure if make files would work though. 
Regards

**Comment by s1nka** [comment]
On Eclipse same errors

**Comment by cr0ssmind** [comment]
Hello,

I have started porting the whole project into Cmake:
https://tree.taiga.io/project/cr0ssmind-openmohaa/epic/1

Hopefully I can push first version soon.
cheers

**Comment by inukaze** [comment]
Hi there reading the README file well i had download Eclipse-CPP
The Readme had [this link](http://www.eclipse.org/downloads/packages/eclipse-ide-cc-developers/marsr) from 2015

Following the readme, i installed eclipse, i set the workspace path to openmohaa/misc/nix then import that same folder path

![openmohaa](https://user-images.githubusercontent.com/1647332/80772416-b29df000-8b24-11ea-9e5d-c33e9fac3324.png)

Then use Project -> Clean -> Clean All
Then use Project -> Build All

![openmohaa2](https://user-images.githubusercontent.com/1647332/80772422-b467b380-8b24-11ea-8634-e14839c5e5b1.png)


Well @cr0ssmind when you had the first version for test, public here too :D

**Comment by Ley0k** [comment]
Try building using the CMakeLists.txt included at the root of the project

**Comment by inukaze** [comment]
> Try building using the CMakeLists.txt included at the root of the project

How ?

**Comment by s1nka** [comment]
> > Try building using the CMakeLists.txt included at the root of the project
> 
> How ?

in run folder run:

```
cmake ./
```

but in Debian i get error from make:
```
[  0%] Building CXX object CMakeFiles/fgame.dir/code/qcommon/class.cpp.o
In file included from /tmp/openmohaa-master/code/qcommon/con_set.h:28,
                 from /tmp/openmohaa-master/code/qcommon/class.h:28,
                 from /tmp/openmohaa-master/code/qcommon/class.cpp:25:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:65:14: error: explicit specialization in non-namespace scope ‘class block_s<aclass, blocksize>’
   65 |     template<>
      |              ^
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:66:12: error: too few template-parameter-lists
   66 |     struct selectType_t<8> {
      |            ^~~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:69:14: error: explicit specialization in non-namespace scope ‘class block_s<aclass, blocksize>’
   69 |     template<>
      |              ^
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:70:12: error: too few template-parameter-lists
   70 |     struct selectType_t<16> {
      |            ^~~~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:73:14: error: explicit specialization in non-namespace scope ‘class block_s<aclass, blocksize>’
   73 |     template<>
      |              ^
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:74:12: error: too few template-parameter-lists
   74 |     struct selectType_t<32> {
      |            ^~~~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:77:14: error: explicit specialization in non-namespace scope ‘class block_s<aclass, blocksize>’
   77 |     template<>
      |              ^
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:78:12: error: too few template-parameter-lists
   78 |     struct selectType_t<64> {
      |            ^~~~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<unsigned int, script_label_t>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<unsigned int, script_label_t>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<unsigned int, script_label_t>’
/tmp/openmohaa-master/code/game/gamescript.h:77:39:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<unsigned int, script_label_t>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<unsigned int, script_label_t>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<unsigned int, Container<SafePtr<SimpleEntity> > >, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<unsigned int, Container<SafePtr<SimpleEntity> > >, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<unsigned int, Container<SafePtr<SimpleEntity> > >’
/tmp/openmohaa-master/code/game/world.h:48:35:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<unsigned int, Container<SafePtr<SimpleEntity> > >, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<unsigned int, Container<SafePtr<SimpleEntity> > >, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<ScriptVariable, ScriptVariable>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<ScriptVariable, ScriptVariable>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<ScriptVariable, ScriptVariable>’
/tmp/openmohaa-master/code/qcommon/con_set.h:138:24:   required from ‘class con_map<ScriptVariable, ScriptVariable>’
/tmp/openmohaa-master/code/script/scriptvariable.h:75:44:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<ScriptVariable, ScriptVariable>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<ScriptVariable, ScriptVariable>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/con_set.h: In instantiation of ‘class Entry<ScriptVariable, ScriptVariable>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:88:44:   required from ‘struct block_s<Entry<ScriptVariable, ScriptVariable>, 256>::info_t’
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:93:12:   required from ‘class block_s<Entry<ScriptVariable, ScriptVariable>, 256>’
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<ScriptVariable, ScriptVariable>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<ScriptVariable, ScriptVariable>’
/tmp/openmohaa-master/code/qcommon/con_set.h:138:24:   required from ‘class con_map<ScriptVariable, ScriptVariable>’
/tmp/openmohaa-master/code/script/scriptvariable.h:75:44:   required from here
/tmp/openmohaa-master/code/qcommon/con_set.h:49:57: error: ‘Entry<k, v>::key’ has incomplete type
   49 |         k                                               key;
      |                                                         ^~~
In file included from /tmp/openmohaa-master/code/qcommon/class.cpp:26:
/tmp/openmohaa-master/code/qcommon/listener.h:37:7: note: forward declaration of ‘class ScriptVariable’
   37 | class ScriptVariable;
      |       ^~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/con_set.h:50:57: error: ‘Entry<k, v>::value’ has incomplete type
   50 |         v                                               value;
      |                                                         ^~~~~
/tmp/openmohaa-master/code/qcommon/listener.h:37:7: note: forward declaration of ‘class ScriptVariable’
   37 | class ScriptVariable;
      |       ^~~~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<short3, ScriptVariable>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<short3, ScriptVariable>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<short3, ScriptVariable>’
/tmp/openmohaa-master/code/script/scriptvariable.h:280:36:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<short3, ScriptVariable>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<short3, ScriptVariable>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<unsigned int, GameScript*>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<unsigned int, GameScript*>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<unsigned int, GameScript*>’
/tmp/openmohaa-master/code/qcommon/con_set.h:138:24:   required from ‘class con_map<unsigned int, GameScript*>’
/tmp/openmohaa-master/code/game/scriptmaster.h:65:37:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<unsigned int, GameScript*>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<unsigned int, GameScript*>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<str, str>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<str, str>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<str, str>’
/tmp/openmohaa-master/code/qcommon/con_arrayset.h:29:7:   required from ‘class con_arrayset<str, str>’
/tmp/openmohaa-master/code/game/scriptmaster.h:70:27:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<str, str>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<str, str>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<unsigned int, unsigned int>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<unsigned int, unsigned int>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<unsigned int, unsigned int>’
/tmp/openmohaa-master/code/qcommon/class.cpp:442:21:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<unsigned int, unsigned int>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<unsigned int, unsigned int>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<Event*, EventDef>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<Event*, EventDef>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<Event*, EventDef>’
/tmp/openmohaa-master/code/qcommon/con_set.h:138:24:   required from ‘class con_map<Event*, EventDef>’
/tmp/openmohaa-master/code/qcommon/class.cpp:693:42:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<Event*, EventDef>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<Event*, EventDef>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h: In instantiation of ‘class block_s<Entry<command_t, command_t>, 256>’:
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:178:11:   required from ‘class MEM_BlockAlloc<Entry<command_t, command_t>, 256>’
/tmp/openmohaa-master/code/qcommon/con_set.h:190:31:   required from ‘class con_set<command_t, command_t>’
/tmp/openmohaa-master/code/qcommon/con_arrayset.h:29:7:   required from ‘class con_arrayset<command_t, command_t>’
/tmp/openmohaa-master/code/qcommon/class.cpp:729:26:   required from here
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:82:11: error: invalid use of incomplete type ‘struct block_s<Entry<command_t, command_t>, 256>::selectType_t<16>’
   82 |     using offset_t = typename selectType_t<bitsNeeded>::type;
      |           ^~~~~~~~
/tmp/openmohaa-master/code/qcommon/mem_blockalloc.h:63:12: note: declaration of ‘struct block_s<Entry<command_t, command_t>, 256>::selectType_t<16>’
   63 |     struct selectType_t;
      |            ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/fgame.dir/build.make:76: CMakeFiles/fgame.dir/code/qcommon/class.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/fgame.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by smallmodel** [comment]
Use a clang version that has full support for C++17 as stated in the README. Clang version 11 should work but I didn't that one. Clang-13 worked for me

**Comment by smallmodel** [comment]
Btw the errors are actually a g++ [bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85282)

**Comment by smallmodel** [comment]
Pushed a fix, the project should now compile with both Clang and GCC.

**Comment by inukaze** [comment]
$ lsb_release -a
LSB Version:	1.4
Distributor ID:	slackware
Description:	Slackware Linux 
Release:	14.2
Codename:	n/a

$ cd /tmp

$ git clone https://github.com/openmoh/openmohaa ; cd openmohaa
Clonando en 'openmohaa'...
remote: Enumerating objects: 5101, done.
remote: Counting objects: 100% (2294/2294), done.
remote: Compressing objects: 100% (1396/1396), done.
remote: Total 5101 (delta 924), reused 2180 (delta 884), pack-reused 2807
Recibiendo objetos: 100% (5101/5101), 57.73 MiB | 114.00 KiB/s, listo.
Resolviendo deltas: 100% (2225/2225), listo.
Actualizando archivos: 100% (2370/2370), listo.

$ cmake ./
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.10 or higher is required.  You are running version 3.5.2


-- Configuring incomplete, errors occurred!

**Comment by DraGo0oN** [comment]
At least CMake 3.10 is rquired.
Upgrade your CMake version to 3.10 or higher.
Then go to build folder and run ```cmake ..``` then run ```make```

**Comment by s1nka** [comment]
> Pushed a fix, the project should now compile with both Clang and GCC.

Thanks. Build fine with gcc or clang

**Comment by smallmodel** [comment]
You should now be able to compile the project using CMake 3.5.

**Comment by inukaze** [comment]
`rm -rf /tmp/openmohaa`
`cd /tmp`
`git clone https://github.com/openmoh/openmohaa ; cd openmohaa`

> Clonando en 'openmohaa'...
> remote: Enumerating objects: 5168, done.
> remote: Counting objects: 100% (2361/2361), done.
> remote: Compressing objects: 100% (1431/1431), done.
> remote: Total 5168 (delta 972), reused 2228 (delta 916), pack-reused 2807
> Recibiendo objetos: 100% (5168/5168), 57.84 MiB | 89.00 KiB/s, listo.
> Resolviendo deltas: 100% (2273/2273), listo.
> Actualizando archivos: 100% (2370/2370), listo

`cmake ./`

> -- The C compiler identification is GNU 5.5.0
> -- The CXX compiler identification is GNU 5.5.0
> -- Check for working C compiler: /usr/bin/cc
> -- Check for working C compiler: /usr/bin/cc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Architecture detected: x86_64, suffix set to x86_64
> -- Using Unix naming convention
> CMake Error at CMakeLists.txt:91 (target_compile_features):
>   target_compile_features specified unknown feature "cxx_std_11" for target
>   "fgame".
> 
> 
> CMake Error at CMakeLists.txt:181 (target_compile_features):
>   target_compile_features specified unknown feature "cxx_std_11" for target
>   "omohaaded".
> 
> 
> -- Looking for pthread.h
> -- Looking for pthread.h - found
> -- Looking for pthread_create
> -- Looking for pthread_create - not found
> -- Looking for pthread_create in pthreads
> -- Looking for pthread_create in pthreads - not found
> -- Looking for pthread_create in pthread
> -- Looking for pthread_create in pthread - found
> -- Found Threads: TRUE  
> -- Configuring incomplete, errors occurred!
> See also "/tmp/openmohaa/CMakeFiles/CMakeOutput.log".
> See also "/tmp/openmohaa/CMakeFiles/CMakeError.log".

i forgot i should how i can check  cxx_std and which version is my system using right now.
i remember some years ago trying to compile something i force the system use c++11 and not c++98


Well i attach the log files generated by the output process : 
[CMakeError.log](https://github.com/openmoh/openmohaa/files/10683757/CMakeError.log)
[CMakeOutput.log](https://github.com/openmoh/openmohaa/files/10683758/CMakeOutput.log)

**Comment by smallmodel** [comment]
Fixed generation issue with CMake 3.5.2, tested with this version, now it's working

**Comment by inukaze** [comment]
> Fixed generation issue with CMake 3.5.2, tested with this version, now it's working

```
rm -rf /tmp/openmohaa
cd /tmp
git clone https://github.com/openmoh/openmohaa ; cd openmohaa
cmake ./
make
TARGET_DIR="/home/inukaze/Escritorio/openmohaa"
mkdir -p "$TARGET_DIR"
cp "omohaadedx86_64" "$TARGET_DIR/"
cp "gamex86_64.so" "$TARGET_DIR/"
cd "$TARGET_DIR/"
chmod 777 -R "$TARGET_DIR"
```

i copy my "main" sub-folder to the "$TARGET_DIR" and start the binary executable generated by make (i think is just for "dedicated server") :

`./omohaadedx86_64 `

> --- Common Initialization ---
> 0.49-alpha (for 1.12) linux-x86_64-debug Feb  9 2023
> ----- FS_Startup -----
> Current search path:
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak6Es.pk3 (471 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak5.pk3 (259 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak4.pk3 (593 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak3.pk3 (669 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak2.pk3 (4722 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak1.pk3 (396 files)
> /media/Compartido/Inukaze/LinuxHome/Escritorio/openmohaa/main/Pak0.pk3 (11174 files)
>  
> ----------------------
> 18284 files in pk3 files
> execing default.cfg
> execing menu.cfg
> execing newconfig.cfg
> ]STUB: not implemented in /tmp/openmohaa/code/sys/sys_unix.c line 1027.
> STUB: not implemented in /tmp/openmohaa/code/sys/sys_unix.c line 1027.
> Config: openmohaa.cfg
> ]STUB: wtf in /tmp/openmohaa/code/sys/sys_unix.c line 1049.
> STUB: wtf in /tmp/openmohaa/code/sys/sys_unix.c line 1060.
> couldn't exec configs/openmohaa.cfg
> couldn't exec localized.cfg
> execing autoexec.cfg
> couldn't exec custom.cfg
> You are now setup for easy mode.
> ]STUB: not implemented in /tmp/openmohaa/code/sys/sys_unix.c line 1027.
> --- Common Initialization Complete --- 1936 ms
> -- Localization: I see 0 localization files
> --- Localization: reading file global/localization.txt
> Loading Localization File global/localization.txt
> --- Error: Invalid Entry... RefName: Go on and run, you yellow-bellies!		LocName: �Continuar corriendo, gordufos!  Attempting to Recover.
> --- Warning: Recovered From Error, Data May have been lost.
> Loaded 1135 localization entries
> Opening IP socket: localhost:12203
> ]

Note : i had the spanish version of this videogame.

**Comment by smallmodel** [comment]
That's good.
There is no client version, only dedicated server version as stated by the *README.md* file. You can try to enter in the dedicated server console `g_gametype 1;devmap dm/mohdm4` then start a Win32 MOH:AA client and connect to the server (*localhost:12203*) that running in Slackware
